### PR TITLE
Multiple updates to steps in: CAS - Generate Unique ID | Synthetic Data Generation 

### PR DESCRIPTION
**Pull request by SundareshSankaran**
# Please include answers to these questions as part of your pull request

In the GitHub webUI, use the Write tab to modify the Markdown text that is part of the pull request. For each question simply place an X inside the square brackets, \[X\], that represents your answer. Make sure there are no blanks inside the brackets, otherwise MarkDown doesn't render properly. Using the Preview tab while editing this form, you can see the formatted/rendered version of the message.

- **Q1: Confirm that you have the right to submit the code that is being contributed. Please consider the origin of your code and confirm you have the appropriate rights to make the submission subject to the Apache 2.0 license that applies to everything in this repository of custom steps. If so, follow the instructions for the [Contributor Agreement](ContributorAgreement.txt) (which is based on the industry-standard Developer Certificate of Origin (DCO)).**
   - [X] Yes, I have the right to submit the contributed code on behalf of myself, my company, or any other owner of the code. I have also attached my signed copy of the DCO to this message.
   - [ ] No
- **Q2: Confirm that your contribution does not include any personally identifiable information (PII), for example, in any examples used in your README file.**
   - [X] My contribution does NOT include PII data
   - [ ] My contribution includes PII data
- **Q3: Confirm your contribution does not include any encryption or other export-controlled technology.**
   - [X] My contribution does NOT contain encryption or other export-controlled technology
   - [ ] My contribution includes encryption or other export-controlled technology
[ContributorAgreement.txt](https://github.com/sassoftware/sas-studio-custom-steps/files/12362754/ContributorAgreement.txt)

### CODE REVIEW

**Comment by SundareshSankaran** [comment]
To avoid queuing, my request is if we could address these changes in a consolidated manner as they involve common type of updates and edits.  A quick summary:

**CAS - Generate Unique ID**

	1. Added a trigger variable to control execution of this step in run time.
	2. Refactored About Page
	3. Comma separated subparameters.

**SDG - Train a Synthetic Data Generator**

	1. Changed variable selectors
	2. Nominal variables added
	3. Added Minibatch size option
	4. Added GPU option
	5. Refactored Program section
	6. Comma separated subparameters.
	7. Runtime control
        8. Refactored About page

File: Synthetic Data Generation/README.md [hunk 1]
```diff
@@ -21,18 +21,52 @@ Tested in Viya 4, Stable 2022.09
...
-3. Optional - training parameters such as number of epochs etc. (Advanced)
+3. Optional - additional training parameters such as number of epochs, the minibatch size, and whether to use Graphical Processing Units (GPUs). (Advanced tab)
 4. Input port - attach a training dataset
 5. Output ports - attach tables for the sample observations and the desired model binary (astore).
 
 ![Train a Synthetic Data Generator](./img/train-a-synthetic-data-generator.png)
 
+#### Run-time control
+
+**Note that this is optional.**
```
Comment (snlwih) [inline Synthetic Data Generation/README.md:38]: Add > at beginning of line to emphasize this note and visually separate it more clearly from the section header. 
So this line would read like this:

> **Note that this is optional.**

File: CAS - Generate Unique ID/README.md [hunk 2]
```diff
@@ -31,14 +31,48 @@ Tested in Viya 4, Stable 2022.11
 
 Note that all columns are copied over to the output table.
 
+#### Run-time control
+
+**Note that this is optional.**  In some scenarios, you may wish to dynamically control whether this custom step runs or simply "passes through" without doing anything, in a SAS Studio session. The following macro variable is set to initialize with a value of 1 by default, indicating an "enabled" status and allowing the custom step to run.
```
Comment (snlwih) [inline CAS - Generate Unique ID/README.md:36]: Add > at beginning of line to emphasize this note and visually separate it more clearly from the section header. And separate the note line from the actual text
So this line would now become three lines and read like this:

> Note that this is optional.

In some scenarios, you may wish to dynamically control whether this custom step runs or simply "passes through" without doing anything, in a SAS Studio session. The following macro variable is set to initialize with a value of 1 by default, indicating an "enabled" status and allowing the custom step to run.

File: Synthetic Data Generation/README.md [hunk 3]
```diff
@@ -78,5 +146,27 @@ proc datasets library=sampsio; run;
 
 ## Change Log
 
-Version 1.0 (06OCT2022)
-
+- Version 1.1 (16AUG2023):
```
Comment (snlwih) [inline Synthetic Data Generation/README.md:149]: This list does not render properly caused by

- a missing minus (-) sign in front of the subsection headers.
- a mixture of using tabs and spaces for indenting, although I'm not 100% sure that will cause rendering issues. Still use one approach only, and as there are only a few tabs in this file I suggest to replace the tab-character with spaces. 

So the lines for the 16AUG update would roughly read like this:

- Version 1.1 (16AUG2023):
  -  **Train a Synthetic Data Generator** (updated)
       1. UI: variable selectors added
       2. UI: nominal input variables added
       3. ...

**Review by snlwih — CHANGES_REQUESTED** [review]
Have added comments in the Synthetic Data Generation\README.md as some of its content does not render properly on GitHub.

File: Synthetic Data Generation/README.md (hunk 1 above)
Comment (SundareshSankaran) [reply Synthetic Data Generation/README.md:38]: Done

File: Synthetic Data Generation/README.md (hunk 3 above)
Comment (SundareshSankaran) [reply Synthetic Data Generation/README.md:149]: I believe I have come to a workable compromise.  The lettered list seems to be giving me trouble, and so I have gone ahead with Roman number-based.  Tabs are now replaced with spaces.  This seems like some one-off problem where some other reason's making things look ugly.

**Comment by SundareshSankaran** [comment]
Thanks for the comments, @snlwih .  I've addressed the individual items in within-file comments (if that helps).

**Comment by snlwih** [comment]
Thanks. I noticed that Synthetic Data Generation\README.md has two occurences of "Note that this is optional", and I only marked one of them.  This is not important and I will merge it as-is.

**Comment by snlwih** [comment]
SAS Internal Due Diligence Review - COMPLETED on 22AUG2023

**Comment by snlwih** [comment]
Note to self: 

- After reviewing the changes and resolving the comments, and before merging, I should have used the **Review Changes** button at the top right of the **Files changed** tab to indicate that all the requested changes had been implemented. 
- I only did this after I had performed the **merge pull request**. 
- Performing the Review Changes task resulted in the following:
    - The big red box "changes requested" disappeared
    - A comment box being added in this **Conversation** tab (which shows just above this comment box), saying "Finished review". That comment box has a link **View reviewed changes" that displays all the comments in all files from both the reviewer and the contributor.

**Pull request merged**